traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/ds/LetsEncryptDnsChallengeWatcher.java

```diff
@@ -0,0 +1,173 @@
+package com.comcast.cdn.traffic_control.traffic_router.core.ds;
+
+/*
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.comcast.cdn.traffic_control.traffic_router.core.config.ConfigHandler;
+import com.comcast.cdn.traffic_control.traffic_router.core.util.AbstractResourceWatcher;
+import com.comcast.cdn.traffic_control.traffic_router.core.util.JsonUtils;
+import com.comcast.cdn.traffic_control.traffic_router.core.util.JsonUtilsException;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.log4j.Logger;
+
+import java.io.*;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+
+public class LetsEncryptDnsChallengeWatcher extends AbstractResourceWatcher {
+    private static final Logger LOGGER = Logger.getLogger(LetsEncryptDnsChallengeWatcher.class);
+    public static final String DEFAULT_LE_DNS_CHALLENGE_URL = "https://${toHostname}/api/1.4/letsencrypt/dnsrecords/";
+    private static final String configFile = "/opt/traffic_router/db/cr-config.json";
+
+    private ConfigHandler configHandler;
+
+    public void setConfigHandler(final ConfigHandler configHandler) {
+        this.configHandler = configHandler;
+    }
+
+    public LetsEncryptDnsChallengeWatcher() {
+        setDatabaseUrl(DEFAULT_LE_DNS_CHALLENGE_URL);
+    }
+
+    @Override
+    @SuppressWarnings("PMD.CyclomaticComplexity")
```

Comment:
I think maybe this means you should just break this method apart

Reply:
done